src/client.cpp

```diff
@@ -393,7 +393,14 @@ status_t Client::Impl::uploadBackground(id_t accountId, id_t instrumentId,
 //    -F "data=@/path/to/image.png;type=image/png"    cs->addFormField(kStrKey, kStrFLIPBOOK);
     cs->addFormField(kStrKey, kStrBACKGROUND);
     cs->addFormField(kStrTimestamp, toIsoTimeString(timestamp));
-    cs->addFormFile(kStrData, imageFile, mimeTypeFromFilePath(imageFile));
+
+    if (payload.data)
+        cs->addFormFile(kStrData, payload.data, payload.dataSize, payload.mimeType.c_str());
```

Comment:
We need to make sure we avoid Base64 encoding which brings 30% size penalty.
If we do not, we shall think how to do it without it.
See also some discussion here.http://blog.marcinbudny.com/2014/02/sending-binary-data-along-with-rest-api.html#.WHgXwKIrKIZ
If we have base64 in current impl, we probably shall go first way in this article.
We need to change Connect API on web backend, but since we are asked to be super efficient, we have to do it earlier or later. So, we shall either do it or prepare well for it.
Lets discuss.